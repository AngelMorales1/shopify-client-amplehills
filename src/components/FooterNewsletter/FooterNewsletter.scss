@import 'settings';

.FooterNewsletter {
  margin-top: spacing(28);
  flex-direction: column;

  @media ($medium-up) {
    flex-direction: row;
  }

  &__container {
    flex-direction: column;
    width: 100%;

    @media ($medium-up) {
      flex-direction: row;
      width: 50%;
      justify-content: center;
    }
  }

  &__text-container {
    @media ($medium-up) {
      margin-right: spacing(6);
    }
  }

  &__help-text {
    top: spacing(-4);
  }

  &__text {
    font-family: $font-carter;
    font-size: $font-6;
  }

  &__text-field {
    width: 100%;

    @media ($medium-up) {
      width: 60%;
      margin-right: spacing(6);
    }

    input {
      padding-top: calc(#{spacing(4)} + 0.1875rem);
      padding-bottom: calc(#{spacing(4)} - 0.1875rem);
    }
  }
}
