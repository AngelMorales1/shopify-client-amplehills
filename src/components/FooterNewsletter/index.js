import React from 'react';
import cx from 'classnames';
import styles from './FooterNewsletter.scss';
import RoutesWithoutNewsletter from 'constants/RoutesWithoutNewsletter';
import { Image, Button, TextField } from 'components/base';

const FooterNewsletter = () => {
  const routeOmitsNewsletter = () => {
    return RoutesWithoutNewsletter.includes(this.props.pathname);
  };

  if (routeOmitsNewsletter()) return null;
  return (
    <div
      className={cx(
        styles['FooterNewsletter'],
        'my4 py4 px3 flex items-center'
      )}
    >
      <div
        className={cx(
          styles['FooterNewsletter__container'],
          'flex items-center'
        )}
      >
        <p
          className={cx(
            styles['FooterNewsletter__text'],
            'my2 text-peach center'
          )}
        >
          Sign up for the latest news from us!
        </p>
        <Image
          className="my2 xs-hide sm-hide icon"
          alt="Decorative arrow"
          src="/assets/images/arrow-straight-right.svg"
        />
      </div>
      <div
        className={cx(
          styles['FooterNewsletter__container'],
          'flex items-center'
        )}
      >
        <TextField
          className={cx(styles['FooterNewsletter__text-field'], 'my2 pb1')}
          placeholder="Enter your email address"
          variant="madison-blue-outline-round"
        />
        <Button
          className="my2 px3"
          label="Sign Up"
          variant="primary"
          color="madison-blue"
        />
      </div>
    </div>
  );
};

export default FooterNewsletter;
