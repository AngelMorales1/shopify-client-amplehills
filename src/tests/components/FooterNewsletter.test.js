import React from 'react';
import { shallow } from 'enzyme';

import FooterNewsletter from 'components/FooterNewsletter';

it('renders without data', () => {
  const component = shallow(<FooterNewsletter />);

  expect(component).toMatchSnapshot();
});

it('renders with data', () => {
  const component = shallow(<FooterNewsletter pathname={'/foo'} />);

  expect(component).toMatchSnapshot();
});
