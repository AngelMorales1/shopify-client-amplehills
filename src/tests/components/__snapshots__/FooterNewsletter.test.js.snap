// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FooterNewsletter
    pathname="/foo"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="flex items-center relative"
>
          <div
                    className="flex flex-column justify-center items-center my2"
          >
                    <p
                              className="small carter center absolute"
                    >
                              Get the inside scoop on new flavors, discounts and more.
                    </p>
                    <p
                              className="text-peach center"
                    >
                              Sign up for our newsletter!
                    </p>
          </div>
          <Image
                    alt="Decorative arrow"
                    bg={false}
                    className="my2 xs-hide sm-hide icon"
                    onImgLoad={[Function]}
                    src="/assets/images/arrow-straight-right.svg"
                    style={Object {}}
          />
</div>,
        <MailchimpSubscribe
          render={[Function]}
          url="https://amplehills.us9.list-manage.com/subscribe/post?u=45ed3f1291b1f51d971f051ff&id=9783303dfa"
/>,
      ],
      "className": "py4 px3 flex items-center z-1",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="flex flex-column justify-center items-center my2"
>
              <p
                            className="small carter center absolute"
              >
                            Get the inside scoop on new flavors, discounts and more.
              </p>
              <p
                            className="text-peach center"
              >
                            Sign up for our newsletter!
              </p>
</div>,
            <Image
              alt="Decorative arrow"
              bg={false}
              className="my2 xs-hide sm-hide icon"
              onImgLoad={[Function]}
              src="/assets/images/arrow-straight-right.svg"
              style={Object {}}
/>,
          ],
          "className": "flex items-center relative",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p
                  className="small carter center absolute"
>
                  Get the inside scoop on new flavors, discounts and more.
</p>,
                <p
                  className="text-peach center"
>
                  Sign up for our newsletter!
</p>,
              ],
              "className": "flex flex-column justify-center items-center my2",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Get the inside scoop on new flavors, discounts and more.",
                  "className": "small carter center absolute",
                },
                "ref": null,
                "rendered": "Get the inside scoop on new flavors, discounts and more.",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Sign up for our newsletter!",
                  "className": "text-peach center",
                },
                "ref": null,
                "rendered": "Sign up for our newsletter!",
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "alt": "Decorative arrow",
              "bg": false,
              "children": null,
              "className": "my2 xs-hide sm-hide icon",
              "onImgLoad": [Function],
              "src": "/assets/images/arrow-straight-right.svg",
              "style": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "render": [Function],
          "url": "https://amplehills.us9.list-manage.com/subscribe/post?u=45ed3f1291b1f51d971f051ff&id=9783303dfa",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="flex items-center relative"
>
            <div
                        className="flex flex-column justify-center items-center my2"
            >
                        <p
                                    className="small carter center absolute"
                        >
                                    Get the inside scoop on new flavors, discounts and more.
                        </p>
                        <p
                                    className="text-peach center"
                        >
                                    Sign up for our newsletter!
                        </p>
            </div>
            <Image
                        alt="Decorative arrow"
                        bg={false}
                        className="my2 xs-hide sm-hide icon"
                        onImgLoad={[Function]}
                        src="/assets/images/arrow-straight-right.svg"
                        style={Object {}}
            />
</div>,
          <MailchimpSubscribe
            render={[Function]}
            url="https://amplehills.us9.list-manage.com/subscribe/post?u=45ed3f1291b1f51d971f051ff&id=9783303dfa"
/>,
        ],
        "className": "py4 px3 flex items-center z-1",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="flex flex-column justify-center items-center my2"
>
                <p
                                className="small carter center absolute"
                >
                                Get the inside scoop on new flavors, discounts and more.
                </p>
                <p
                                className="text-peach center"
                >
                                Sign up for our newsletter!
                </p>
</div>,
              <Image
                alt="Decorative arrow"
                bg={false}
                className="my2 xs-hide sm-hide icon"
                onImgLoad={[Function]}
                src="/assets/images/arrow-straight-right.svg"
                style={Object {}}
/>,
            ],
            "className": "flex items-center relative",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p
                    className="small carter center absolute"
>
                    Get the inside scoop on new flavors, discounts and more.
</p>,
                  <p
                    className="text-peach center"
>
                    Sign up for our newsletter!
</p>,
                ],
                "className": "flex flex-column justify-center items-center my2",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Get the inside scoop on new flavors, discounts and more.",
                    "className": "small carter center absolute",
                  },
                  "ref": null,
                  "rendered": "Get the inside scoop on new flavors, discounts and more.",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Sign up for our newsletter!",
                    "className": "text-peach center",
                  },
                  "ref": null,
                  "rendered": "Sign up for our newsletter!",
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "alt": "Decorative arrow",
                "bg": false,
                "children": null,
                "className": "my2 xs-hide sm-hide icon",
                "onImgLoad": [Function],
                "src": "/assets/images/arrow-straight-right.svg",
                "style": Object {},
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "render": [Function],
            "url": "https://amplehills.us9.list-manage.com/subscribe/post?u=45ed3f1291b1f51d971f051ff&id=9783303dfa",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`renders without data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FooterNewsletter
    pathname=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="flex items-center relative"
>
          <div
                    className="flex flex-column justify-center items-center my2"
          >
                    <p
                              className="small carter center absolute"
                    >
                              Get the inside scoop on new flavors, discounts and more.
                    </p>
                    <p
                              className="text-peach center"
                    >
                              Sign up for our newsletter!
                    </p>
          </div>
          <Image
                    alt="Decorative arrow"
                    bg={false}
                    className="my2 xs-hide sm-hide icon"
                    onImgLoad={[Function]}
                    src="/assets/images/arrow-straight-right.svg"
                    style={Object {}}
          />
</div>,
        <MailchimpSubscribe
          render={[Function]}
          url="https://amplehills.us9.list-manage.com/subscribe/post?u=45ed3f1291b1f51d971f051ff&id=9783303dfa"
/>,
      ],
      "className": "py4 px3 flex items-center z-1",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="flex flex-column justify-center items-center my2"
>
              <p
                            className="small carter center absolute"
              >
                            Get the inside scoop on new flavors, discounts and more.
              </p>
              <p
                            className="text-peach center"
              >
                            Sign up for our newsletter!
              </p>
</div>,
            <Image
              alt="Decorative arrow"
              bg={false}
              className="my2 xs-hide sm-hide icon"
              onImgLoad={[Function]}
              src="/assets/images/arrow-straight-right.svg"
              style={Object {}}
/>,
          ],
          "className": "flex items-center relative",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p
                  className="small carter center absolute"
>
                  Get the inside scoop on new flavors, discounts and more.
</p>,
                <p
                  className="text-peach center"
>
                  Sign up for our newsletter!
</p>,
              ],
              "className": "flex flex-column justify-center items-center my2",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Get the inside scoop on new flavors, discounts and more.",
                  "className": "small carter center absolute",
                },
                "ref": null,
                "rendered": "Get the inside scoop on new flavors, discounts and more.",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Sign up for our newsletter!",
                  "className": "text-peach center",
                },
                "ref": null,
                "rendered": "Sign up for our newsletter!",
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "alt": "Decorative arrow",
              "bg": false,
              "children": null,
              "className": "my2 xs-hide sm-hide icon",
              "onImgLoad": [Function],
              "src": "/assets/images/arrow-straight-right.svg",
              "style": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "render": [Function],
          "url": "https://amplehills.us9.list-manage.com/subscribe/post?u=45ed3f1291b1f51d971f051ff&id=9783303dfa",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="flex items-center relative"
>
            <div
                        className="flex flex-column justify-center items-center my2"
            >
                        <p
                                    className="small carter center absolute"
                        >
                                    Get the inside scoop on new flavors, discounts and more.
                        </p>
                        <p
                                    className="text-peach center"
                        >
                                    Sign up for our newsletter!
                        </p>
            </div>
            <Image
                        alt="Decorative arrow"
                        bg={false}
                        className="my2 xs-hide sm-hide icon"
                        onImgLoad={[Function]}
                        src="/assets/images/arrow-straight-right.svg"
                        style={Object {}}
            />
</div>,
          <MailchimpSubscribe
            render={[Function]}
            url="https://amplehills.us9.list-manage.com/subscribe/post?u=45ed3f1291b1f51d971f051ff&id=9783303dfa"
/>,
        ],
        "className": "py4 px3 flex items-center z-1",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="flex flex-column justify-center items-center my2"
>
                <p
                                className="small carter center absolute"
                >
                                Get the inside scoop on new flavors, discounts and more.
                </p>
                <p
                                className="text-peach center"
                >
                                Sign up for our newsletter!
                </p>
</div>,
              <Image
                alt="Decorative arrow"
                bg={false}
                className="my2 xs-hide sm-hide icon"
                onImgLoad={[Function]}
                src="/assets/images/arrow-straight-right.svg"
                style={Object {}}
/>,
            ],
            "className": "flex items-center relative",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p
                    className="small carter center absolute"
>
                    Get the inside scoop on new flavors, discounts and more.
</p>,
                  <p
                    className="text-peach center"
>
                    Sign up for our newsletter!
</p>,
                ],
                "className": "flex flex-column justify-center items-center my2",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Get the inside scoop on new flavors, discounts and more.",
                    "className": "small carter center absolute",
                  },
                  "ref": null,
                  "rendered": "Get the inside scoop on new flavors, discounts and more.",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Sign up for our newsletter!",
                    "className": "text-peach center",
                  },
                  "ref": null,
                  "rendered": "Sign up for our newsletter!",
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "alt": "Decorative arrow",
                "bg": false,
                "children": null,
                "className": "my2 xs-hide sm-hide icon",
                "onImgLoad": [Function],
                "src": "/assets/images/arrow-straight-right.svg",
                "style": Object {},
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "render": [Function],
            "url": "https://amplehills.us9.list-manage.com/subscribe/post?u=45ed3f1291b1f51d971f051ff&id=9783303dfa",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
